dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -1036,6 +1038,29 @@ public void syncUserMetadataFromGitHub(User user) {
         }
     }
 
+    public boolean syncUserMetadataFromGitHubByUsername(User user) {
+        // eGit user object
+        try {
+            GHUser ghUser = github.getUser(user.getUserProfiles().get(TokenType.GITHUB_COM.toString()).username);
+            User.Profile profile = new User.Profile();
```

Comment:
Seems like lines 1045 through 1056 could be a function shared with 1024 through 1034 above